nestedtemplates/configuringVMAzureDiskEncryption-kek.json

```diff
@@ -1,94 +1,117 @@
+                "forceUpdateTag": "[parameters('vmEncryptionSettings').sequenceVersion]",
+                "settings": {
+                    "AADClientID": "[parameters('aadClientID')]",
+                    "KeyVaultURL": "[parameters('vmEncryptionSettings').keyVaultURL]",
+                    "KeyEncryptionKeyURL": "[parameters('keyEncryptionKeyURL')]",
+                    "KeyEncryptionAlgorithm": "[parameters('vmEncryptionSettings').keyEncryptionAlgorithm]",
+                    "VolumeType": "[parameters('vmEncryptionSettings').volumeType]",
+                    "EncryptionOperation": "[parameters('vmEncryptionSettings').encryptionOperation]"
                 },
-                "secretUrl": "[reference(resourceId('Microsoft.Compute/virtualMachines/extensions',  parameters('vmName'), parameters('vmEncryptionSettings').diskEncryptionExtensionName)).instanceView.statuses[0].message]"
-              },
-              "keyEncryptionKey": {
-                "sourceVault": {
-                  "id": "[parameters('vmEncryptionSettings').keyVaultResourceID]"
-                },
-                "keyUrl": "[parameters('keyEncryptionKeyURL')]"
-              }
+                "protectedSettings": {
+                    "AADClientSecret": "[parameters('aadClientSecret')]"
+                }
             }
-          }
+        },
+        {
+            "apiVersion": "2016-04-30-preview",
+            "type": "Microsoft.Compute/virtualMachines",
+            "name": "[parameters('vmName')]",
+            "location": "[resourceGroup().location]",
+            "tags": {
+                "Project": "BluePrint"
+            },
+            "dependsOn": [
+                "[resourceId('Microsoft.Compute/virtualMachines/extensions',  parameters('vmName'), parameters('vmEncryptionSettings').diskEncryptionExtensionName)]"
+            ],
+            "properties": {
+                "storageProfile": {
+                    "osDisk": {
+                        "encryptionSettings": {
+                            "diskEncryptionKey": {
+                                "sourceVault": {
+                                    "id": "[parameters('vmEncryptionSettings').keyVaultResourceID]"
+                                },
+                                "secretUrl": "[reference(resourceId('Microsoft.Compute/virtualMachines/extensions',  parameters('vmName'), parameters('vmEncryptionSettings').diskEncryptionExtensionName)).instanceView.statuses[0].message]"
+                            },
+                            "keyEncryptionKey": {
+                                "sourceVault": {
+                                    "id": "[parameters('vmEncryptionSettings').keyVaultResourceID]"
+                                },
+                                "keyUrl": "[parameters('keyEncryptionKeyURL')]"
+                            }
+                        }
+                    }
+                }
+            },
+            "resources": [{
```

Comment:
Not commented out.  This may cause issue, if so comment out just as in *-nokek.json